Src/VimCore/CommandUtil.fs

```diff
@@ -1819,12 +1819,12 @@ type internal CommandUtil
             | OperationKind.CharacterWise -> x.CaretPoint
             | OperationKind.LineWise -> x.CaretLine.Start
 
-        x.PutCore point stringData registerValue.OperationKind moveCaretAfterText
+        x.PutCore point stringData stringData registerValue.OperationKind moveCaretAfterText
 
     /// Put the contents of the specified register after the cursor.  Used for the
     /// normal 'p', 'gp', 'P', 'gP', ']p' and '[p' commands.  For linewise put operations
     /// the point must be at the start of a line
-    member x.PutCore point stringData operationKind moveCaretAfterText =
+    member x.PutCore point originalStringData stringData operationKind moveCaretAfterText =

```

Comment:
Think we could simplify this a bit. After looking at this `originalStringData` and `stringData` only differ in the case that `stringData` is `StringData.Simple` instance and that's only useful in positioning the caret below. Suggestion I had is the following:

1. Move this to `Option string` and use a name along the lines of  `originalSimpleText`. 
1. Use this value to create a `StringData` to match on: 

``` fsharp
let m = 
  match originalSimpleText
  | Some s -> StringData.Simple s
  | None -> stringData
```

Even then the text is still duplicated between `originalSimpleText` and `stringData` and there is an implicit assumption they are equal. I can't really think of a great way around that though. Other than making the new parameter a `bool` and doing a more complex match.

Reply:
The bool ended up being the clearest way to indicate precisely what was desired, i.e "move the caret as if the string data were simple." Thanks for the suggestion to improve it.